tests/integration/hive_metastore/test_migrate.py

```diff
@@ -564,6 +563,33 @@ def test_migrate_external_tables_with_principal_acl_aws(
     assert match
 
 
+@retried(on=[NotFound], timeout=timedelta(minutes=3))
+def test_migrate_external_tables_with_principal_acl_aws_warehouse(
+    ws, make_user, prepared_principal_acl, make_warehouse_permissions, make_warehouse, env_or_skip
+):
+    if not ws.config.is_aws:
+        pytest.skip("temporary: only works in aws test env")
+    ctx, table_full_name, _, _ = prepared_principal_acl
+    ctx.with_dummy_resource_permission()
+    warehouse = make_warehouse()
+    table_migrate = ctx.tables_migrator
+    user = make_user()
+    make_warehouse_permissions(
```

Comment:
Do I understand correctly that we are migrating user access to locations to which it has access via access to a warehouse? Where do we mount or connect locations to a warehouse?

Reply:
Yes that is right. we dont need to connect location to the warehouse, that's done at the UC level. So if a user has access to a table in non uc (through the spn configured in the warehouse), that user will have access to the location and the table once it is migrated to UC, warehouse spn is not used in UC world